awx/api/validators.py

```diff
@@ -0,0 +1,69 @@
+import re
+
+from django.core.validators import RegexValidator, MaxLengthValidator, validate_ipv46_address
+from django.core.exceptions import ValidationError
+
+
+class HostnameRegexValidator(RegexValidator):
+    """
+    Fully validates a domain name that is compliant with norms in Linux/RHEL
+        - Cannot start with a hyphen
+        - Cannot begin with, or end with a "."
+        - Cannot contain any whitespaces
+        - Entire hostname is max 255 chars (including dots)
+        - Each domain/label is between 1 and 63 characters, except top level domain, which must be at least 2 characters
+        - Supports ipv4, ipv6, simple hostnames and FQDNs
+        - Follows RFC 9210 (modern RFC 1123, 1178) requirements
+
+    Accepts an IP Address or Hostname as the argument
+    """
+
+    def __init__(self, *args, **kwargs):
+
+        if 'regex' not in kwargs.keys():
+            self.regex = '^[a-z0-9][-a-z0-9]*$|^([a-z0-9][-a-z0-9]{0,62}[.])*[a-z0-9][-a-z0-9]{1,62}$'
```

Comment:
I think the `super` call will wind up not playing well with this (and the flags below), at least under some circumstances.

Maybe just do a fresh dict with the regex and flags, then `new_kwargs.update(kwargs)`?

Reply:
Well, ok, thinking it through a bit more I think it won't matter